In [2]:
import pandas as pd

df = pd.read_csv('vgsales.csv') #loading the dataset
df.head()

,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
0,1,Wii Sports,Wii,2006.0,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74
1,2,Super Mario Bros.,NES,1985.0,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24
2,3,Mario Kart Wii,Wii,2008.0,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82
3,4,Wii Sports Resort,Wii,2009.0,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00
4,5,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37


In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  str    
 2   Platform      16598 non-null  str    
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  str    
 5   Publisher     16540 non-null  str    
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), str(4)
memory usage: 1.4 MB


In [4]:
df.describe()

,Rank,Year,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
count,16598.000000,16327.000000,16598.000000,16598.000000,16598.000000,16598.000000,16598.000000
mean,8300.605254,2006.406443,0.264667,0.146652,0.077782,0.048063,0.537441
std,4791.853933,5.828981,0.816683,0.505351,0.309291,0.188588,1.555028
min,1.000000,1980.000000,0.000000,0.000000,0.000000,0.000000,0.010000
25%,4151.250000,2003.000000,0.000000,0.000000,0.000000,0.000000,0.060000
50%,8300.500000,2007.000000,0.080000,0.020000,0.000000,0.010000,0.170000
75%,12449.750000,2010.000000,0.240000,0.110000,0.040000,0.040000,0.470000
max,16600.000000,2020.000000,41.490000,29.020000,10.220000,10.570000,82.740000


Time to see if dataset is clean

In [5]:
df.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [6]:
df.dtypes

Rank              int64
Name                str
Platform            str
Year            float64
Genre               str
Publisher           str
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

In [7]:
df.columns

Index(['Rank', 'Name', 'Platform', 'Year', 'Genre', 'Publisher', 'NA_Sales',
       'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales'],
      dtype='str')

In [8]:
df.shape

(16598, 11)

In [9]:
df.duplicated().sum()

np.int64(0)

So from the above stuff we found that there are 16k+ rows, 11 columns
As for the missing values
Year has 271
Publisher, 58
We got no duplicates

Year somehow is float64, it should be int, lets change that
Before that lets make a copy of our dataframe

In [24]:
cleaned_df = df.copy()
cleaned_df["Year"] = df["Year"].astype("Int64")

In [12]:
cleaned_df["Year"].dtype
df["Year"].dtype

dtype('float64')

In [13]:
cleaned_df["Year"].dtype

Int64Dtype()

Nice hehe
Now to handle missing year values as well as mussing publisher values

In [14]:
cleaned_df["Publisher"] = cleaned_df["Publisher"].fillna("Unknown")

In [16]:
cleaned_df["Publisher"].isna().sum()

np.int64(0)

coolio

In [17]:
cleaned_df.isna().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher         0
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

Now for the year

In [18]:
cleaned_df["Year"] = cleaned_df["Year"].fillna(cleaned_df["Year"].median())

In [19]:
cleaned_df.isna().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

Onto the next

In [20]:
cleaned_df.dtypes

Rank              int64
Name                str
Platform            str
Year              Int64
Genre               str
Publisher           str
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

# Lets move onto the 8 Business Questions
### Q1. Which 10 games generated the highest global sales?

In [25]:
top_games = cleaned_df[["Name", "Global_Sales"]].sort_values(by="Global_Sales", ascending=False).head(10)

In [26]:
top_games

,Name,Global_Sales
0,Wii Sports,82.74
1,Super Mario Bros.,40.24
2,Mario Kart Wii,35.82
3,Wii Sports Resort,33.00
4,Pokemon Red/Pokemon Blue,31.37
5,Tetris,30.26
6,New Super Mario Bros.,30.01
7,Wii Play,29.02
8,New Super Mario Bros. Wii,28.62
9,Duck Hunt,28.31


Wii Sports is the highest selling game in the dataset, followed by Super Mario Bros and Mario Kart Wii

### Q2. If a publisher could only develop for one historical platform, which platform would have offered the largest market opportunity?

In [27]:
platform_sales = (cleaned_df.groupby("Platform")["Global_Sales"].sum().sort_values(ascending=False))

In [28]:
platform_sales

Platform
PS2     1255.64
X360     979.96
PS3      957.84
Wii      926.71
DS       822.49
PS       730.66
GBA      318.50
PSP      296.28
PS4      278.10
PC       258.82
XB       258.26
GB       255.45
NES      251.07
3DS      247.46
N64      218.88
SNES     200.05
GC       199.36
XOne     141.06
2600      97.08
WiiU      81.86
PSV       61.93
SAT       33.59
GEN       28.36
DC        15.97
SCD        1.87
NG         1.44
WS         1.42
TG16       0.16
3DO        0.10
GG         0.04
PCFX       0.03
Name: Global_Sales, dtype: float64

PS2 generated the highest lifetime sales among all gaming platforms. -w-

### Q3. Which game genre is most profitable globally? 
interesting question .o. lets see

In [31]:
genre_sales = (cleaned_df.groupby("Genre")["Global_Sales"].sum().sort_values(ascending=False))

In [32]:
genre_sales

Genre
Action          1751.18
Sports          1330.93
Shooter         1037.37
Role-Playing     927.37
Platform         831.37
Misc             809.96
Racing           732.04
Fighting         448.91
Simulation       392.20
Puzzle           244.95
Adventure        239.04
Strategy         175.12
Name: Global_Sales, dtype: float64

Action games generate the highest total global revenue, that makes sense XD

### Q4. Who are the dominant players in the video game industry based on total sales?

In [34]:
pub_sales = (cleaned_df.groupby("Publisher")["Global_Sales"].sum().sort_values(ascending=False))

pub_sales.head(10)

Publisher
Nintendo                        1786.56
Electronic Arts                 1110.32
Activision                       727.46
Sony Computer Entertainment      607.50
Ubisoft                          474.72
Take-Two Interactive             399.54
THQ                              340.77
Konami Digital Entertainment     283.64
Sega                             272.99
Namco Bandai Games               254.09
Name: Global_Sales, dtype: float64

As expected, its Nintendo

### Q5. Which games sold more than 10 million copies globally?

In [42]:
ten_mil_copies = cleaned_df[cleaned_df["Global_Sales"] > 10].sort_values(by="Global_Sales", ascending=False)[["Name", "Platform", "Global_Sales"]]

In [44]:
ten_mil_copies.head(10)

,Name,Platform,Global_Sales
0,Wii Sports,Wii,82.74
1,Super Mario Bros.,NES,40.24
2,Mario Kart Wii,Wii,35.82
3,Wii Sports Resort,Wii,33.00
4,Pokemon Red/Pokemon Blue,GB,31.37
5,Tetris,GB,30.26
6,New Super Mario Bros.,DS,30.01
7,Wii Play,Wii,29.02
8,New Super Mario Bros. Wii,Wii,28.62
9,Duck Hunt,NES,28.31


Only a small number of titles cross the 10-million-sales mark :/

### Q6: How many games were released for the PlayStation 2 (PS2) platform?

In [46]:
ps2_count = cleaned_df[cleaned_df['Platform'] == 'PS2'].shape[0]
ps2_count

2161

### Q7: Which games resonated more strongly with European audiences than North American audiences?

In [52]:
cleaned_df[cleaned_df['EU_Sales'] > cleaned_df['NA_Sales']].sort_values('Global_Sales', ascending = False).head(5)[['Name', 'EU_Sales', 'NA_Sales', 'Global_Sales']]

,Name,EU_Sales,NA_Sales,Global_Sales
10,Nintendogs,11.00,9.07,24.76
16,Grand Theft Auto V,9.27,7.01,21.40
19,Brain Age: Train Your Brain in Minutes a Day,9.26,4.75,20.22
27,Brain Age 2: More Training in Minutes a Day,5.36,3.44,15.30
33,Call of Duty: Black Ops 3,5.81,5.77,14.24


The games are Nintendogs, GTA V, Brain Age, Brain Age 2 as well as COD:BO3

### Q8: What is the average global sale for action games vs strategy games?

In [49]:
avg_action = cleaned_df[cleaned_df['Genre'] == 'Action']['Global_Sales'].mean()
avg_strategy = cleaned_df[cleaned_df['Genre'] == 'Strategy']['Global_Sales'].mean()
print(f"Average Global Sales   Action: {avg_action:.2f}M, Strategy: {avg_strategy:.4f}M")

Average Global Sales   Action: 0.53M, Strategy: 0.2572M


Average Global 
Sales Action: 0.53M, Strategy: 0.2572M 
### Done! :>